### ISSUE TITLE
pytest fails tests/test_place_seqs.py 

### ISSUE THREAD

**Issue by gabridinosauro** [picrust/picrust2#312]
Hello,

I am trying to install picrust2. I am following the instructions to install from source, as form conda it gets stuck in the exectuing transaction point.

With installing it from source, I manage to install it correctly. Nevertheless the test pytest fails.
The error output is very long, so I just take the parts that I think to be significant.
It happens in both my personal computer (macbook pro intel with the updated version of conda and mamba) and in the HPC of my university. So I do not think it is a matter of machine/set up.

Thanks in advance!

Gabri

```
=================================================================================== test session starts ====================================================================================
platform linux -- Python 3.8.16, pytest-7.3.1, pluggy-1.0.0
rootdir: /xdisk/pkiela/schiro/picrust2-2.5.2
plugins: cov-4.1.0
collected 61 items                                                                                                                                                                         

tests/test_hsp.py ........                                                                                                      [ 13%]
tests/test_metagenome_pipeline.py ...............                                                                               [ 37%]
tests/test_pathway_pipeline.py ....                                                                                             [ 44%]
tests/test_place_seqs.py ........F                                                                                              [ 59%]
tests/test_util.py ....................                                                                                         [ 91%]
tests/test_workflow.py .....                                                                                                    [100%]

```

SECOND PART:

 ```
-------------------------------------------------------- Captured stdout call ---------------------------------------------------------

-------------------------------------------------------- Captured stderr call ---------------------------------------------------------
hmmalign --trim --dna --mapali /xdisk/pkiela/schiro/picrust2-2.5.2/picrust2/default_files/prokaryotic/pro_ref/pro_ref.fna --informat FASTA -o /tmp/tmpevs9xtw2/query_align.stockholm /xdisk/pkiela/schiro/picrust2-2.5.2/picrust2/default_files/prokaryotic/pro_ref/pro_ref.hmm /xdisk/pkiela/schiro/picrust2-2.5.2/tests/test_data/place_seqs/study_seqs_test.fasta

All raw input sequences were the same length (400)

run_sepp.py --tree /xdisk/pkiela/schiro/picrust2-2.5.2/picrust2/default_files/prokaryotic/pro_ref/pro_ref.tre --raxml /xdisk/pkiela/schiro/picrust2-2.5.2/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info --cpu 1 --molecule dna --outdir /tmp/tmpevs9xtw2/sepp_out -seed 297834 --alignment /xdisk/pkiela/schiro/picrust2-2.5.2/picrust2/default_files/prokaryotic/pro_ref/pro_ref.fna --fragment /tmp/tmpevs9xtw2/study_seqs_filtered.fasta

Error running this command:
run_sepp.py --tree /xdisk/pkiela/schiro/picrust2-2.5.2/picrust2/default_files/prokaryotic/pro_ref/pro_ref.tre --raxml /xdisk/pkiela/schiro/picrust2-2.5.2/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info --cpu 1 --molecule dna --outdir /tmp/tmpevs9xtw2/sepp_out -seed 297834 --alignment /xdisk/pkiela/schiro/picrust2-2.5.2/picrust2/default_files/prokaryotic/pro_ref/pro_ref.fna --fragment /tmp/tmpevs9xtw2/study_seqs_filtered.fasta

Standard error of the above failed command:
Traceback (most recent call last):
  File "/home/u1/schiro/.conda/envs/picrust2/bin/run_sepp.py", line 3, in <module>
    from sepp.exhaustive import ExhaustiveAlgorithm
  File "/home/u1/schiro/.conda/envs/picrust2/lib/python3.8/site-packages/sepp/exhaustive.py", line 6, in <module>
    from sepp.algorithm import AbstractAlgorithm
  File "/home/u1/schiro/.conda/envs/picrust2/lib/python3.8/site-packages/sepp/algorithm.py", line 11, in <module>
    from sepp.tree import PhylogeneticTree
  File "/home/u1/schiro/.conda/envs/picrust2/lib/python3.8/site-packages/sepp/tree.py", line 25, in <module>
    from dendropy.datamodel.treemodel import _convert_node_to_root_polytomy as \
ImportError: cannot import name '_convert_node_to_root_polytomy' from 'dendropy.datamodel.treemodel' (/home/u1/schiro/.conda/envs/picrust2/lib/python3.8/site-packages/dendropy/datamodel/treemodel/__init__.py)
```

**Comment by gavinmdouglas** [comment]
Thanks for pointing out this error. It looks like there is an incompatibility with SEPP and newer versions of dendropy. Since this is an error with sepp specifically, if you want to use the default epa-ng read placement rather than sepp you can ignore this issue. Otherwise you will need to specifically specify what versions of dendropy and sepp to install, but it will take some troubleshooting to figure out what versions are compatible.

Cheers,

Gavin

**Comment by ArnaudBelcour** [comment]
Hello,

I also encounter this issue when testing my installation of picrust2. I do some testing with dendropy versions (by installing/removing it with pip) and it seems that the issue arises with dendropy version `4.6.0`. By using the previous version available in pip (version `4.5.2`), there is no error.
So I write this answer if someone wants to use sepp.

This issue has also been reported to sepp developers: https://github.com/smirarab/sepp/issues/125.

Best regards,
Arnaud.

**Comment by Jeef059** [comment]
> Hello,
> 
> I also encounter this issue when testing my installation of picrust2. I do some testing with dendropy versions (by installing/removing it with pip) and it seems that the issue arises with dendropy version `4.6.0`. By using the previous version available in pip (version `4.5.2`), there is no error. So I write this answer if someone wants to use sepp.
> 
> This issue has also been reported to sepp developers: [smirarab/sepp#125](https://github.com/smirarab/sepp/issues/125).
> 
> Best regards,  

Hello sir, I seem to have encountered the same problem when installing PICRUSt2, but after I saw your reply, I tried to use pip to install dendropy to version 4.5.2, but another error was reported. My sepps is version 4.3.10, and I would like to ask which version of dendropy should be used. Or both versions need to be changed.

**Comment by R-Wright-1** [comment]
Hi there, this works for me with the fix given in #330: `pip install dendropy==4.5.2`

Could you give some more details on what the error is that you get? 

Robyn

**Comment by R-Wright-1** [comment]
I'm just closing this issue as there hasn't been any further response.
